spec/models/message_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+
+RSpec.describe Message, type: :model do
+  subject { create(:message) }
+
+  describe 'validations' do
+    it { should validate_presence_of(:text) }
+
+    describe 'user does not blongs to chat users' do
+      let(:user_a) { create(:user) }
+      let(:user_b) { create(:user) }
+      let!(:chat_a) { create(:chat, user_a: user_a) }
+      let(:test) { create(:message, chat: chat_a, user: user_b) }
+
+      it 'does not create message' do
+        msg = 'Validation failed: Message User must be the same than one of the chat users'
+        expect { test }.to raise_error(ActiveRecord::RecordInvalid, msg)
+      end
```

Comment:
mmm are you not using subject here ?

maybe you can do something like 
`subject { build(:message, chat: chat_a, user: user_b) } `
and then validate the following
```
subject.valid?
expect(subject.errors.messages).to include(error_message)
```

WDYT?